llvm/lib/Transforms/InstCombine/InstCombineMulDivRem.cpp

```diff
@@ -905,6 +905,16 @@ Instruction *InstCombinerImpl::visitFMul(BinaryOperator &I) {
     }
   }
 
+  // (X * 0.0) * constant => X * 0.0
+  Constant *C1;
+  if (match(Op0, m_FMul(m_Value(X), m_Constant(C1))) &&
+      match(C1, m_AnyZeroFP()) && match(Op1, m_Constant(C))) {
```

Comment:
The match of C1 to m_AnyZeroFP is redundant, you can just directly check m_AnyZeroFP where you first matched C1

Reply:
`m_AnyZeroFP` doesn't capture so you can't write `m_AnyZeroFP(C1)`. But it probably should.

Reply:
> It can't point at a specific constant in the non-splat case.

I'd expect it to point at the whole vector-typed `Constant`.